### DISCUSSION TITLE
Create lifetime token for use in CD

### DISCUSSION THREAD

**Discussion by carlos-mayorga** [envault/envault#8] [Ideas]
**Is your feature request related to a problem? Please describe it**
I am currently using a CD with Codeship in my project and I would always like to update my environment variables, in codeship view "Uh oh! Looks like your setup token is invalid, please get another!"

**Describe the solution you'd like**
Use a lifetime token

**Comment by danharrin** [comment]
Hey, after you set up for the first time, you can use the `npx envault` command on its own up sync again.

Does this fix your issue?

**Comment by carlos-mayorga** [comment]
"Does this fix your issue?"

sorry no, on each CI/CD is new context

**Comment by danharrin** [comment]
Is there any way to persist credentials between CI runs with Codeship?

**Comment by carlos-mayorga** [comment]
in codeship use enviroments

![alt text](https://i.imgur.com/moFB6i3.png "Logo Title Text 1")

**Comment by danharrin** [comment]
Are these variables accessible through `process.env` in Node?

If they are, we could use that as an alternative configuration method.

**Comment by simonschaufi** [comment]
How long is the `authToken` valid in the file `.envault.json`?

**Comment by danharrin** [comment]
Hey @simonschaufi, there's no expiry on the `authToken`, however it will cease to work if the user account it was created by is deleted.

**Reply by wedi** [reply]
I'm having a very similar same use case. It would be great if one could create such persistent API keys with description and optionally expiry date (and maybe even restricted to selected apps) via GUI just like it's common on GitHub, Slack etc. See it in action here: https://github.com/settings/tokens

**Reply by danharrin** [reply]
Hey @wedi, after initial setup, your `.envault.json` file will contain a permanent API token for all future updates. I'd suggest making this file available for your CD, giving it permanent API access for .env updates.

**Reply by wedi** [reply]
I understood that but I'd like to have an overview over which tokens exist and when the last usage was. It's not really viable to recreate user accounts for API token revocation. And for usage in CI one had to run the envault client locally, grab the API key from the file and transfer it to the CI environment variables as persisting a secrect that got created during a CI run is not viable, too. This in turn is prone to spread API keys over multiple systems.

**Comment by aaronhuisinga** [comment]
We'd love to see this added as well. Our CD pipeline doesn't let us define environment variables, so we'd like to use Envault to pull them in for testing and deployment. Persisting the `authToken` to an `.envault.json` file isn't as option for us either as we'd rather not add that into version control (the CD just syncs our Github repo). 

Since our CD uses a different container for each build, it would be great if we could define a long term revokable token somewhere.
